### PR TITLE
make sure path is actually normalized in all regards

### PR DESCRIPTION
**Pull request by pfitzseb**
Fixes https://github.com/JuliaGizmos/AssetRegistry.jl/issues/7.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -21,9 +21,21 @@ using JSON
         -e 'using AssetRegistry; AssetRegistry.register(pwd());'
         ```)
 
-    @test JSON.parse(String(read(joinpath(homedir(), ".jlassetregistry.json")))) == Dict(AssetRegistry.filekey(key) => [pwd(), 1])
+    # @test JSON.parse(String(read(joinpath(homedir(), ".jlassetregistry.json")))) == Dict(AssetRegistry.filekey(key) => [pwd(), 1])
```
[the remark this answers is not in the document]
Comment (pfitzseb) [reply test/runtests.jl]: No, I just forgot to uncomment that after I figured out why it broke.

**Comment by pfitzseb** [comment]
Hm, it does not actually fix the issue I was having. That _is_ fixed on InteractBase master though. Can you tag a new release, @piever?

**Comment by piever** [comment]
Do you need also [this PR](https://github.com/piever/InteractBase.jl/pull/110) to be merged to fix your issue?

**Comment by pfitzseb** [comment]
No, InteractBase#master works for me.
When I tested this PR yesterday I was on InteractBase master (instead of release).

**Pull request merged**